src/ngraph/op/fused/stack.cpp

```diff
@@ -0,0 +1,102 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <memory>
+#include <numeric>
+
+#include "matmul.hpp"
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/op/concat.hpp"
+#include "ngraph/op/fused/stack.hpp"
+#include "ngraph/op/reshape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::Stack::type_info;
+
+op::Stack::Stack(const OutputVector& args, int64_t axis)
+    : FusedOp(OutputVector{args})
+    , m_axis(axis)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::Stack::Stack(const NodeVector& args, int64_t axis)
+    : Stack(as_output_vector(args), axis)
+{
+}
+
+shared_ptr<Node> op::Stack::copy_with_new_args(const NodeVector& new_args) const
+{
+    return make_shared<Stack>(new_args, m_axis);
+}
+
+void op::Stack::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    ngraph_error("Not Yet Implemented");
+}
+
+void op::Stack::pre_validate_and_infer_types()
```

Comment:
This is optional but it may be better to override validate_and_infer_types() instead of pre_validate_and_infer_types() like here https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/frontend/fluid/operators/reduce_sum.cpp#L74-L85